Framework/DataHandling/src/LoadMuonNexusV2Helper.cpp

```diff
@@ -154,6 +154,23 @@ double loadTimeZeroFromNexusFile(const NeXus::NXEntry &entry) {
   }
 }
 
+std::vector<double> loadTimeZeroListFromNexusFile(const NeXus::NXEntry &entry) {
+  NXClass det_class = entry.openNXGroup(NeXusEntry::DETECTOR);
+  NXInt spectrum_index = det_class.openNXInt("spectrum_index");
+  spectrum_index.load();
+  auto ndets = spectrum_index.dim0();
+
+  NXDouble timeZeroClass = det_class.openNXDouble(NeXusEntry::TIMEZERO);
+  std::vector<double> timeZeroVector = timeZeroClass.vecBuffer();
+  if (timeZeroVector.size() == 0) {
```

Comment:
should we check that the length of the vector is what we expect (i.e. ndets=length)?

Reply:
The timezero entry can either be a single value or an array. This is a check to distinguish between these two cases.

Reply:
yes, but what if there are only 24 time zeros and 94 detectors? I think we should check that too and if they do not match give a warning and do something else (maybe use the average for all)?

Reply:
I have added in a check that the length of the time zero vector matches the number of spectra. As I don't think this should ever happen I have thrown a runtime error if this occurs to warn the user their nexus file is invalid.

Reply:
do we have a way to say just give a single file so the file is usable?

Reply:
We could always put a number in here but I don't think we should. The format specifies that the time zero should either be a single value or an array of the same length as the number of spectra. If we have corrupted data I think the correct solution is to inform the user the file is wrong. Any scheme we come up with to replace the missing data is likely to be wrong in some way and may just to serve to hide the fact that the file is incorrect.